### PR TITLE
Better support for regexp

### PR DESCRIPTION
**Pull request by dlmr**
This PR will add support for the following in regards to RegExp:
- Validation function to be used with settings, commands and actions/hooks
- Convertor that can be used with commands and automatically with settings

Some tests for this is also added along with tests that where missing for `isPromise`.

### CODE REVIEW

File: test/validation/validators.js [hunk 1]
```diff
@@ -135,6 +137,46 @@ describe('roc', () => {
...
+            describe('isPromise', () => {
+                it('should return info object if requested', () => {
+                    expect(isPromise(null, true))
+                        .toEqual({
+                            type: 'Promise',
+                            required: false
+                        });
+                });
+
+                it('should validate a RegExp correctly', () => {
```
Comment (andreasrs) [inline test/validation/validators.js]: Promise

File: test/validation/validators.js [hunk 2]
```diff
@@ -135,6 +137,46 @@ describe('roc', () => {
...
+                    expect(isPromise(Promise.resolve()))
+                        .toBe(true);
+                });
+
+                it('should return error if value is not a RegExp', () => {
```
Comment (andreasrs) [inline test/validation/validators.js]: Promise, not a RegExp

**Comment by andreasrs** [comment]
Minor things in tests. The description text.

Kudos for implementing tests here, i guess it helps as the change-sets are smaller

:+1:

File: test/validation/validators.js (hunk 1 above)
Comment (dlmr) [inline test/validation/validators.js]: Should be RegExp here.

**Pull request merged**